app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt

```diff
@@ -113,43 +104,12 @@ object TelemetryWrapper {
         // are readable/writable.
         val threadPolicy = StrictMode.allowThreadDiskWrites()
         try {
-            val telemetryEnabled = DataUploadPreference.isEnabled(context)
-
-            val configuration = TelemetryConfiguration(context)
-                    .setServerEndpoint("https://incoming.telemetry.mozilla.org")
-                    .setAppName(TELEMETRY_APP_NAME_FOCUS_TV)
-                    .setUpdateChannel(BuildConfig.BUILD_TYPE)
-                    .setPreferencesImportantForTelemetry(
-                            Settings.TRACKING_PROTECTION_ENABLED_PREF,
-                            TelemetrySettingsProvider.PREF_CUSTOM_HOME_TILE_COUNT,
-                            TelemetrySettingsProvider.PREF_TOTAL_HOME_TILE_COUNT
-                    )
-                    .setSettingsProvider(TelemetrySettingsProvider(context))
-                    .setCollectionEnabled(telemetryEnabled)
-                    .setUploadEnabled(telemetryEnabled)
-
-            val serializer = JSONPingSerializer()
-            val storage = FileTelemetryStorage(configuration, serializer)
-            val client = HttpURLConnectionTelemetryClient()
-            val scheduler = JobSchedulerTelemetryScheduler()
-
-            TelemetryHolder.set(Telemetry(configuration, storage, client, scheduler)
-                    .addPingBuilder(TelemetryCorePingBuilder(configuration))
-                    .addPingBuilder(TelemetryMobileEventPingBuilder(configuration))
-                    .setDefaultSearchProvider(createDefaultSearchProvider(context)))
+            TelemetryHolder.set(TelemetryFactory.createTelemetry(context))
```

Comment:
> typealias TelemetryEntryPoint = Telemetry and make the factory based on that. EntryPoint isn't a great name so maybe something better.

I think having global typealiases seems like it could get confusing.  I would rather try to open a PR against A-C that changes the class name itself.
> TelemetryHolder -> TelemetryProvider (more conventional)

Unfortunately this is part of A-C.
> TelemetryWrapper -> TelemetryIntegration (to better distinguish from Holder/Provider)

Agreed, and done.

> could also try putting TelemetryWrapper behind a module so the only thing exposed to the app is TelemetryIntegration, simplifying this problem.

Although I haven't needed to yet, keeping TelemetryFactory available and using it in our Application to assemble TelemetryIntegration allows us to substitute in different values for testing.  I think that's worth preserving.